test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1119,46 +1119,59 @@ class test_ucp_sockaddr_wireup_fail : public test_ucp_sockaddr_wireup {
             UCS_TEST_SKIP_R("cannot connect to server");
         }
 
-        if (fail_wireup_type == FAIL_WIREUP_MSG_SEND) {
+        while (!wait(e, wait_ep_flags) && (sender().get_err_num() == 0)) {
+            progress();
+        }
```

Comment:
why not set the callbacks before calling progress?

Reply:
but CM flow would not call ep_am_bcopy anyway

Reply:
setting ep_am_bcopy could be set before `progress()`, but `iface_get_device_address` can't set there

Reply:
before this pr, ep_am_bcopy was set before and iface_get_device_address after
why needed to change?

Reply:
reverted